### PR TITLE
Add new tests for ReadGridded

### LINKED ISSUE #196: ReadGridded.get_var_info_from_files() fails if AUX_REQUIRES.keys() contain *
See `test_ReadGridded_get_var_info_from_files` in #195.
Caused by calling `check_compute_var(conc*`) which expects a variable name with no wildcards.

### PR DESCRIPTION
**Pull request by ejgal**
* Add tests for ReadGridded (#194)
* Fix #196

### CODE REVIEW

File: pyaerocom/io/test/test_readgridded.py [hunk 1]
```diff
@@ -57,34 +58,120 @@ def test_ReadGridded_class_empty():
...
+    assert sorted(r.ts_types) == ['daily', 'monthly']
+
+@testdata_unavail
+def test_ReadGridded_read_var(reader_tm5):
+    r = reader_tm5
+    data = r.read_var('od550aer')
+    assert isinstance(data, GriddedData)
+    ds = data.to_xarray().load()
+    mean = float(ds.mean(dim='time').mean(dim='lat').mean(dim='lon'))
+    npt.assert_almost_equal(mean, 0.09631748)
```
Comment (jgliss) [inline pyaerocom/io/test/test_readgridded.py:84]: Why converting to xarray and doing mean over individual dimensions? You could probably also do data.mean() directly (which will be area weighted, though, by default, but should suffice).

File: pyaerocom/io/test/test_readgridded.py [hunk 2]
```diff
@@ -57,34 +58,120 @@ def test_ReadGridded_class_empty():
...
+        assert r.has_var(var)  # calling has_var
+    assert sorted(r.vars_provided) == sorted(expected)
+    # assert r.has_var('conco3')
+
+
+def test_ReadGridded_prefer_longer():
+    r = ReadGridded(data_dir=path_tm5)
+    daily = r.read_var('abs550aer', ts_type='monthly', flex_ts_type=True,
+                       prefer_longer=True)
+    assert daily.ts_type == 'daily'
```
Comment (jgliss) [inline pyaerocom/io/test/test_readgridded.py:126]: Is there also a monthly field? Consider using parametrize with different settings?

**Review by jgliss — APPROVED** [review]
Good to go, but feel free to check my comments and add some updates.

File: pyaerocom/io/test/test_readgridded.py (hunk 2 above)
Comment (ejgal) [reply pyaerocom/io/test/test_readgridded.py:126]: Yes, abs550aer has daily and monthly files in the testdata.
Running this with ts_type='yearly' and flex_ts_type=True results in `pyaerocom.exceptions.DataQueryError: Could not find file match for query`. Shouldn't flex_ts_type option ensure that it chooses the monthly or daily files?

File: pyaerocom/io/test/test_readgridded.py (hunk 1 above)
Comment (ejgal) [reply pyaerocom/io/test/test_readgridded.py:84]: Forgot that GriddedData has that functionality! The weird way of taking means is caused by some 4D data I worked with recently :)

File: pyaerocom/io/test/test_readgridded.py (hunk 2 above)
Comment (ejgal) [reply pyaerocom/io/test/test_readgridded.py]: Seems that the current behavior is that setting `ts_type` leads to `flex_ts_type` not affecting the behavior.

Comment (jgliss) [reply pyaerocom/io/test/test_readgridded.py]: Hmm, okay, can you make an issue for that?

**Comment by jgliss** [comment]
@ejgal some tests are failing here. Can you check? Let me know if I should revise once more.

**Comment by ejgal** [comment]
Seems that there was an error fetching the testdata. Reran the jobs without changing anything and they passed.

File: pyaerocom/io/test/test_readgridded.py (hunk 2 above)
Comment (ejgal) [reply pyaerocom/io/test/test_readgridded.py]: Yep!

**Pull request merged**